accepted/better-obsoletion/better-obsoletion.md

```diff
@@ -0,0 +1,373 @@
+  suppressing all call sites individually.
+
+* **Documentation pages are easily accessible**. Developers must be able to go
+  from the diagnostic in the error list straight to the documentation that
+  provides the details.
+
+### Non-Goals
+
+* **Detecting specific usage patterns**. Sometimes, an API isn't obsoleted, just
+  a particular combination of arguments or specific values. If we need to
+  identify these, we'll author a custom analyzer.
+
+* **Being able to obsolete APIs in already shipped versions of .NET**. The
+  attribute must be applied to the API surface that the compiler sees and thus
+  must be present in the reference assemblies. We can only obsolete in new
+  versions of the platform. In other words, this doesn't support a sidecar file
+  that can be applied to an existing version of .NET.
+
+* **Being able to remove APIs in later versions**. See
+  [the discussion](#why-is-being-able-to-delete-apis-not-a-goal) in the Q&A
+  section.
+
+* **Hiding all obsoleted APIs from IntelliSense**. Whether or not an API should
+  be hidden is an independent decision. For more details, see
+  [the discussion](#should-obsoleted-members-be-hidden-from-intellisense) in the
+  Q&A section.
+
+## Design
+
+The design proposal piggybacks on the existing `ObsoleteAttribute` that is
+already supported by all .NET compilers. The idea is to expand this a little to
+support the desired scenarios. Alternatives are listed afterwards.
+
+### API Changes
+
+We'd add the following properties to `ObsoleteAttribute`:
+
+```C#
+namespace System
+{
+    public sealed class ObsoleteAttribute : Attribute
+    {
+        // Existing:
+        //
+        // public ObsoleteAttribute();
+        // public ObsoleteAttribute(string message);
+        // public ObsoleteAttribute(string message, bool error);
+        // public bool IsError { get; }
+        // public string Message { get; }
+
+        // New:
+        public string DiagnosticId { get; set; }
+        public string UrlFormat { get; set; }
+        public string Url => UrlFormat == null ? null : string.Format(UrlFormat, DiagnosticId);
+    }
+}
+```
+
+Instead of taking the URL directly, the API takes a format string. This allows
+having a generic URL that includes the diagnostic ID. This avoids having to
+repeat the ID twice and thus making a copy & paste mistake.
+
+### Compiler Changes
+
+The compilers would be enlightened about these new properties and use them as
+follows:
+
+* If `DiagnosticId` is `null`, use the existing diagnostic ID (e.g. `CS0618` in
+  C#). Otherwise, use `DiagnosticId`.
+    - The compiler should honor suppressions for the specified diagnostic id.
+* If `Url` is not `null`, use it as the diagnostic link when rendering them in
+  the IDE.
+* The compiler should assume that `Url` and `DiagnosticId` are independent
+  features, in other words both can be used, either, and neither.
+
+***OPEN ISSUE**: Should the compiler suppress all diagnostics from
```

Comment:
Your formatting since off here making the rest of the text all bold in preview. Perhaps:

```suggestion
**OPEN ISSUE:** *Should the compiler suppress all diagnostics from
```

Reply:
Fixed